### PR TITLE
feat: call v2 endpoints

### PR DESCRIPTION
**Pull request by jmoralez**
* Changes all methods to target the v2 endpoints from the API, which are faster and require smaller payloads.
* Moves the distributed logic to the nixtla_client
* Removes the following deprecated symbols:
  * TimeGPT class
  * NixtlaClient.validate_token
  * short-horizon and long-horizon models

### CODE REVIEW

File: nbs/utils.ipynb [hunk 1]
```diff
@@ -63,6 +63,31 @@
     "def in_colab():\n",
```
Comment (AzulGarza) [inline nbs/utils.ipynb:63]: <p>love this</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/nixtla/pull/429/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

File: nbs/v2.nixtla_client.ipynb [hunk 2]
```diff
@@ -0,0 +1,510 @@
+{
```
Comment (AzulGarza) [inline nbs/v2.nixtla_client.ipynb:1]: <p>should we hide these tests?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/nixtla/pull/429/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (AzulGarza) [inline nbs/v2.nixtla_client.ipynb:1]: <p>nice, should we hide the tests?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/nixtla/pull/429/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -15,34 +15,12 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  nb-sync:
```
Comment (jmoralez) [inline .github/workflows/ci.yaml:18]: This is done in the lint task now

File: nixtla/nixtla_client.py [hunk 4]
```diff
@@ -1,96 +1,62 @@
...
+from utilsforecast.feature_engineering import _add_time_features, time_features
+from utilsforecast.validation import ensure_time_dtype, validate_format, validate_freq
 
-        return wrapper
+if TYPE_CHECKING:
+    import fugue
+    import triad
 
-    return decorator
+from .core.api_error import ApiError
```
Comment (AzulGarza) [inline nixtla/nixtla_client.py:49]: this is very cool

File: nixtla/nixtla_client.py [hunk 5]
```diff
@@ -138,712 +104,434 @@ def wrapper(self, *args, **kwargs):
...
+            ConnectionResetError,
+            httpcore.ConnectError,
+            httpcore.RemoteProtocolError,
+            httpx.ConnectTimeout,
+            httpx.ReadError,
+            httpx.RemoteProtocolError,
+            httpx.ReadTimeout,
+            httpx.PoolTimeout,
+            httpx.WriteError,
+            httpx.WriteTimeout,
```
Comment (AzulGarza) [inline nixtla/nixtla_client.py:120]: awesome

File: nixtla/nixtla_client.py [hunk 6]
```diff
@@ -1262,7 +752,7 @@ def forecast(
             Column that identifies each timestep, its values can be timestamps or integers.
         target_col : str (default='y')
             Column that contains the target.
-        X_df : pandas.DataFrame, optional (default=None)
+        X_df : pandas or polars DataFrame, optional (default=None)
```
Comment (AzulGarza) [inline nixtla/nixtla_client.py:755]: woaaa, very cool that we are supporting polars

**Review by AzulGarza — APPROVED** [review]
lvgtm, awesome work @jmoralez

**Pull request merged**